### ISSUE TITLE
"Further Reading" section in species accounts

### ISSUE THREAD

**Issue by Louis-Backstrom** [bird-team/brisbane-bird-atlas#152]
I was thinking it might be a good idea to add a section to the bottom of species accounts with links to further reading - similar to the bibliography but with stuff that we won't necessarily cite in-text, e.g. the eBird species page, the HBW Alive page, the Cornell Birds of the World page for the species, and other useful links like that - would be a good way to tie the project in with other, broader-scale ones.

Thoughts?

**Comment by jeffreyhanson** [comment]
Yeah, I think that's a great idea! We could update the book to automatically populate this section using links in the settings spreadsheet to make adding this info easier? However, this would limit the flexibility/customizeability of the section meaning that you couldn't add arbitrary text to it? What do you think?

**Comment by Louis-Backstrom** [comment]
I was thinking about that before yeah - because I think there will be many things we want to add "automatically" to every account (e.g. the eBird species page), but there will definitely be times when we want to add arbitrary text to the account as well. It's probably easier to just do it manually as we write the accounts for the moment.

**Comment by jeffreyhanson** [comment]
Yeah, maybe we could split this "Further reading" section into two bits? Since many/most species will have links on the IUCN, the HBW Alive, the Cornell Birds of the World - what if we added (1) a standard/automated section that contains links to these pages and (2)  a more free form optional "Further reading" section which has other links and arbitrary text?

**Comment by jeffreyhanson** [comment]
For section 1, if there are logos for these different projects/sites (e.g. IUCN, the HBW Alive, the Cornell Birds of the World), then we could just have the logos somewhere on a species' account and then redirect the user to the species' account on the project's web page when they click on the logo? Not sure if that's OK with regards to copyright and stuff though.

**Comment by Louis-Backstrom** [comment]
Love the logo idea. I feel like that should be okay copyright wise - logos are usually fine to use? Plus, most of these projects are CC or PD anyway I think. Can definitely check that out though. Could we have the logos appear in like a footer at the bottom of the page?

**Comment by Louis-Backstrom** [comment]
Okay, I tried my hand at creating a sort of banner / footer that we could use to link to various different birding websites - all the individual website logos are in a new folder `/data/logos/` as svgs (1920x1080), plus I combined all 8 (subject to change of course) into one with some decent formatting that could be used:

![all_8](https://user-images.githubusercontent.com/44354406/74100648-1f849800-4b7d-11ea-9e2f-9f929afc2e00.png)

Questions then:

1. @dbl3raf are there any other useful websites you think we might want to link to for all species? The only others I can think of are the EPBC/Aus Gvt pages (e.g. [Magpie Goose](http://www.environment.gov.au/cgi-bin/sprat/public/publicspecies.pl?taxon_id=978)) and Avibase pages but I've excluded both these for the time being as a) I'm not sure they add a huge amount to what we already have b) neither have particularly useful logos (I'm not even sure if Avibase has one formally?) and c) the urls for each species on each site appear to be fairly complex and not conducive to straightforward link creation (which I'm working on right now as well)

2. @jeffreyhanson how easy is it to turn svg(s) into link(s)? Currently I'm working on the premise of populating the `atlas_list.xlsx` file (or a new one should it be needed) with the various taxon identifiers needed to create working links for each of the websites (e.g. for eBird and BotW, Magpie Goose is `maggoo1` and the base link is https://ebird.org/species/ and https://beta.birdsoftheworld.org/bow/species/ (the beta prefix will go once the site properly launches), whereas for IUCN the taxon identifier is `22679732/92826979` and the base link is https://www.iucnredlist.org/species/ - same process hopefully exists for each website). Is it possible then to link parts of the combined svg (ie the section of it that contains the specific website logo) to each of the website links? Alternatively, can all the individual svgs be combined onto each species page in a uniform (nicely-formatted) manner? Hope that's clear enough.

3. Currently I've left the HBW Alive logo out of the combination svg - for two reasons. a) the logo is very messy, on account of me not being able to find a logo for HBW Alive at more than 400px resolution, which didn't convert to svg very well and b) the project is being deprecated and replaced by the Cornell BotW - which we're already linking to (and I believe will be properly launched in upcoming months). @dbl3raf do you think it's necessary to have HBW as a regular link given this?

**Comment by Louis-Backstrom** [comment]
Also - some species will not have pages on all of these websites, either due to taxonomy conflicts (e.g. Australian / Gull-billed Tern might be problematic), project scope (I'm not sure that BLA have species pages for all of our species) or non-species being treated differently by eBird (i.e. domestics, hybrids and the like will only have proper links on the eBird explore page, and probably not any else).

How do we want to treat these special cases?

**Comment by jeffreyhanson** [comment]
Yeah, creating the links is pretty straightforward. Indeed, we would just need (1) some extra columns in the `atlas_list.xlsx` file containing the different identifiers (or URLS see below) for each source, (2) manually insert some R code at the footer of each species' account to tell it to add the logos/links, (3) update the species account template file with the footer too, (4) write an R function that is called at the footer of each page to create the logos/links and write HTML code to automatically arrange them nicely. The main trick is (4), but this shouldn't take more then 5 hours (probably closer to 2--3). The hardest part for me will be writing the HTML code automatically to arrange them nicely, unless all the images are just laid out horizontally (or something like that).

To handle the special cases, I suggest that we write the R code to handle blank cells (NAs) in the spreadsheet. I can write the R code to not show a logo/link for a species if there is a blank cell for it in the spreadsheet. So e.g. if the only the IUCN red list column has a value for a species, then only the logo for the IUCN will be shown. Also, if all of the link columns are blank for a species, then none of the logos will be shown. How does that sound?

To help make the atlas easier to maintain, I suggest that we add the full URLs to the spreadsheet (not just the ID) so that if any part of the IUCN/HBW/etc URLs change in the future then you and Rich can fix this by changing the spreadsheet (ideally with a find-all-replace) and won't need to dive into the code. What do you think?

Also, I wonder what the best way to present the logos/links is? What do you think about implementing them as badges? Eg. like these: https://shields.io/category/other, and we just have them laid out horizontally under the title in the species accounts or at the bottom of the page? If we had images of just the logos (in white so they look nice when overlaid on a grey background), we could create custom badges with the logo on the left side of the badge and the name on the right side of the badge (see  https://github.com/badges/shields/blob/master/doc/logos.md#custom-logos). But this would require some graphics expertise and would probably take a while - so I'm not sure if its worth it?

**Comment by Louis-Backstrom** [comment]
I like the idea of badges, that would make things easier to format I think. Everything else sounds good too, although I don't want to make too much work for you. I've just pushed a commit (ee687fb) that adds a column for the identifier and URL for each species on each of the 8 websites - do you want to try have a go at making something using that?

I'll be adding links/identifiers throughout the day as I work out ways to populate them - hopefully I can get a few species with all 8, some with most and some with very few to give you stuff to work with.

**Comment by Louis-Backstrom** [comment]
> If we had images of just the logos (in white so they look nice when overlaid on a grey background), we could create custom badges with the logo on the left side of the badge and the name on the right side of the badge (see https://github.com/badges/shields/blob/master/doc/logos.md#custom-logos). But this would require some graphics expertise and would probably take a while - so I'm not sure if its worth it?

I've gone and made small square logos out of each of the projects - they should all be easy enough to convert to black and/or white if we need that too - most are already single colour.

**Comment by jeffreyhanson** [comment]
Awesome - thank you very much! I'll take a look at this on the weekend and see what I can do.

**Comment by jeffreyhanson** [comment]
@Louis-Backstrom, just a heads up, I've just pushed a commit with an attempt to add the logos/links to all the species accounts. I had to edit all of the species account files to add an extra bit of code to display the logos/links (termed badges), so you'll probably have to pull the latest version of the repo before you can commit and push any updates.

**Comment by jeffreyhanson** [comment]
Also, the website version might look slightly weird while I'm sorting out the formatting for the logos/links today.

**Comment by jeffreyhanson** [comment]
The BOTW links don't seem to be working for me. For example, the magpie goose URL (https://beta.birdsoftheworld.org/bow/species/maggoo1) shows me this: 

![Access Forbidden - Birds of the World - Google Chrome_013](https://user-images.githubusercontent.com/3610005/74578885-07d56580-4ffc-11ea-8e70-fdcbc379369f.png)

Is this intended?

**Comment by Louis-Backstrom** [comment]
Yes, that's expected - to access the BOTW beta you have to have a Cornell Lab account that has been linked with HBW Alive (or Neotropical Birds / Birds of North America). I believe the full release will also be subscription access only, although they might have a similar "trial" view to HBW Alive, not sure.

For reference, Magpie Goose looks like this to me:

![image](https://user-images.githubusercontent.com/44354406/74579771-ddc67800-4fe8-11ea-9d49-1b361c06d835.png)

**Comment by jeffreyhanson** [comment]
Ok great- thanks for clarifying that! I'm currently playing around with the badges text and page margins to see what looks good. I'll let you know when I think I've found the right mix to see what you think?

**Comment by jeffreyhanson** [comment]
Ok - I think I've tweaked the badges/page width so it looks good - what do you think @Louis-Backstrom and @dbl3raf?

**Comment by Louis-Backstrom** [comment]
Looks good! Feels just a little bit small to me - is there a way we can spread the badges over two lines evenly so that the icons are big enough to see? At least on my screen, they're pretty miniscule:

![image](https://user-images.githubusercontent.com/44354406/74580468-698fd280-4ff0-11ea-9d72-c84d3650e5d3.png)

**Comment by jeffreyhanson** [comment]
Yeah, I agree that some of the icons are pretty hard to see. I'm not sure if enlarging the badges so that they take up two lines is the best approach though. I think they just need to exist so that interested readers can easily access them, and ideally look nice but not take up too much space that they distract from the rest of the species' account? Or is there another purpose that I'm missing?

I think most of the icons look amazing in the badges, and it's just a few that are impacted by the small size. I don't think I can make the icons appear bigger within the badge (I can stretch them out horizontally, but this isn't really what we want). I wonder if it would be possible to simplify/modify the Birds of the World and IUCN logos? And perhaps lighten the blue color in the BirdLife logos so it stands out more?

**Comment by Louis-Backstrom** [comment]
Agreed - I'm wondering if white borders around those two might help pop them out from the grey background? I've just tried a change on each of these - let me know what you think.

**Comment by Louis-Backstrom** [comment]
Are you able to update the badges @jeffreyhanson ?

**Comment by jeffreyhanson** [comment]
Sure - I've just pushed a commit with the new badges. Also, you can update the badges on your computer using the command `make badges`. They will appear in the `assets/badges` folder. You can use settings in the `parameters.yaml` file to control which image files will be used for the badges.

**Comment by Louis-Backstrom** [comment]
Alright, I've pushed the changes I made - they're only minor but I think it helps.

I'm wondering if it's possible to increase the width of the page to help give us more room / make text easier to read? At least on my (16:9) screen, there's a lot of white space on the margins.

![image](https://user-images.githubusercontent.com/44354406/74614269-39684100-5162-11ea-9697-04fbf4f6ec9e.png)

Is it possible in bookdown to decrease the size of these margins?

Also, just noticed that when you change the text size to either extreme, the badge formatting gets messed up - probably a fringe case and not worth debugging too much, but if it's an easy fix, perhaps worth investigating?

![image](https://user-images.githubusercontent.com/44354406/74614287-6583c200-5162-11ea-9bee-30ed5d839bc0.png)
